cloudwatch/cloudwatch.go

```diff
@@ -157,40 +155,52 @@ func NewOutputPlugin(config OutputPluginConfig) (*OutputPlugin, error) {
 	}, nil
 }
 
-func newCloudWatchLogsClient(roleARN string, sess *session.Session, endpoint string, credsEndpoint string, logFormat string) *cloudwatchlogs.CloudWatchLogs {
-	svcConfig := &aws.Config{}
-	if endpoint != "" {
-		defaultResolver := endpoints.DefaultResolver()
-		cwCustomResolverFn := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
-			if service == "logs" {
-				return endpoints.ResolvedEndpoint{
-					URL: endpoint,
-				}, nil
-			}
-			return defaultResolver.EndpointFor(service, region, optFns...)
+func newCloudWatchLogsClient(region, roleARN, cwEndpoint, stsEndpoint, credsEndpoint, logFormat string) (*cloudwatchlogs.CloudWatchLogs, error) {
+	defaultResolver := endpoints.DefaultResolver()
```

Comment:
Can we move this into the `customResolverFn` function definition instead of referencing it via a closure?